File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) 
    {
        int l=0,h=nums.size()-1,n=nums.size();
        // if(n==1)
        // return nums[0];
        while(l<h)
        {
            int mid=l+(h-l)/2;
            // int prev=(mid+n-1)%n;
            // int next=(mid+1)%n;
            
            // if(nums[mid]<=nums[prev] && nums[mid]<=nums[next])
            // return nums[mid];
            
            //move to unsorted part
            if(nums[mid]<nums[h])
            h=mid;
            else //if(nums[mid]<nums[h])
            l=mid+1;
        }
        return nums[l];
    }
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int h=nums.length;
        int l=0,r=h-1;

        while(l<r)
        {
            int mid=l+(r-l)/2;

            if(nums[mid]>nums[r])
            l=mid+1;
            else
            r=mid;
        }
        return nums[l];
    }
}